sdk/src/builder.rs

```diff
@@ -561,6 +569,30 @@ impl Builder {
                 builder.resources.add(id, data)?;
             }
 
+            if file.name().starts_with("ingredient-resources/")
+                && file.name() != "ingredient-resources/"
+            {
+                let mut data = Vec::new();
+                file.read_to_end(&mut data)?;
+
+                let id = file
+                    .name()
+                    .split_once('/')
+                    .map(|(_, second)| second)
+                    .ok_or(Error::BadParam("Invalid resource path".to_string()))?;
+                let format = format_from_path(id)
+                    .ok_or(Error::BadParam("Invalid resource path".to_string()))?;
+                let id = id.replacen(['-'], ":", 1);
+                for ingredient in builder.definition.ingredients.iter_mut() {
+                    let base_id = ingredient.instance_id().to_string();
+                    if id.starts_with(&base_id) {
+                        ingredient
+                            .resources_mut()
+                            .add_with(&base_id, &format, data.clone())?;
```

Comment:
After this line you can add a `break;` and avoid the `.clone()`. Instance IDs are globally unique so the current iterated resource should only match one ingredient thus making the loop safe to break.